open_spiel/python/mfg/algorithms/softmax_policy.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Computes a softmax policy from a value."""
+import numpy as np
+
+from open_spiel.python import policy as policy_std
+from open_spiel.python.mfg import value
+
+
+class SoftmaxPolicy(policy_std.Policy):
+  """Computes the softmax policy of a value."""
+
+  def __init__(self, game, player_ids, temperature,
+               state_action_value: value.ValueFunction,
+               prior_policy: policy_std.Policy = None):
```

Comment:
Since this is optional it is better to mark it as such, i.e. prior_policy: Optional[policy.Policy] = None. You can use

from typing import Optional

in the imports.

Reply:
Done